contracts/EthlanceJobs.sol

```diff
@@ -0,0 +1,851 @@
+pragma solidity 0.5.8;
+pragma experimental ABIEncoderV2;
+
+import "./token/IERC20.sol";
+import "./token/IERC721.sol";
+import "./math/SafeMath.sol";
+
+
+/// @title EthlanceJob
+/// @dev A contract for issuing jobs on Ethereum paying in ETH, ERC20, or ERC721 tokens
+/// @author Mark Beylin <mark.beylin@consensys.net>, Gonçalo Sá <goncalo.sa@consensys.net>, Kevin Owocki <kevin.owocki@consensys.net>, Ricardo Guilherme Schmidt (@3esmit), Matt Garnett <matt.garnett@consensys.net>, Craig Williams <craig.williams@consensys.net>
+contract EthlanceJobs {
+
+  using SafeMath for uint256;
+
+  /*
+   * Structs
+   */
+
+  struct Job {
+    address payable[] issuers; // An array of individuals who have complete control over the job, and can edit any of its parameters
+    address[] approvers; // An array of individuals who are allowed to accept the fulfillments for a particular job
+    address token; // The address of the token associated with the job (should be disregarded if the tokenVersion is 0)
+    uint tokenVersion; // The version of the token being used for the job (0 for ETH, 20 for ERC20, 721 for ERC721)
+    uint balance; // The number of tokens which the job is able to pay out or refund
+    bool hasPaidOut; // A boolean storing whether or not the job has paid out at least once, meaning refunds are no longer allowed
+    Fulfillment[] fulfillments; // An array of Fulfillments which store the various submissions which have been made to the job
+    Contribution[] contributions; // An array of Contributions which store the contributions which have been made to the job
+    address[] hiredCandidates;
+  }
+
+  struct Fulfillment {
+    address payable[] fulfillers; // An array of addresses who should receive payouts for a given submission
+    address submitter; // The address of the individual who submitted the fulfillment, who is able to update the submission as needed
+    uint amount;
+  }
+
+  struct Contribution {
+    address payable contributor; // The address of the individual who contributed
+    uint amount; // The amount of tokens the user contributed
+    bool refunded; // A boolean storing whether or not the contribution has been refunded yet
+  }
+
+  /*
+   * Storage
+   */
+
+  uint public numJobs; // An integer storing the total number of jobs in the contract
+  mapping(uint => Job) public jobs; // A mapping of jobIDs to jobs
```

Comment:
A mapping prevents someone from determining which JobID refers to what if i'm working strictly with the smart contract. This could probably be a listing.

Reply:
@madvas do you think keeping this the same as StandardBounties is more important than changing it for a array?

Reply:
keep as StandardBounties